pkg/generator/caches_test.go

```diff
@@ -185,6 +185,40 @@ func createTestDataForIngress(caches *Caches,
 	_ = caches.SetListeners(kubeClient)
 }
 
+func TestValidateIngress(t *testing.T) {
+	logger := zap.S()
+
+	caches := NewCaches(logger)
+	kubeClient := fake.Clientset{}
+
+	createTestDataForIngress(
+		caches,
+		"ingress_1",
+		"ingress_1_namespace",
+		"cluster_for_ingress_1",
+		"route_for_ingress_1",
+		"internal_host_for_ingress_1",
+		"external_host_for_ingress_1",
+		&kubeClient,
+	)
+
+	translatedIngress := translatedIngress{
+		ingressName:          "ingress_2",
+		ingressNamespace:     "ingress_2_namespace",
+		routes:               []*route.Route{{Name: "route_for_ingress_2"}},
+		clusters:             []*v2.Cluster{{Name: "cluster_for_ingress_2"}},
+		externalVirtualHosts: []*route.VirtualHost{{Name: "external_host_for_ingress_2", Domains: []string{"external_host_for_ingress_2"}}},
+		//This domain should clash with the cached ingress.
+		internalVirtualHosts: []*route.VirtualHost{{Name: "internal_host_for_ingress_2", Domains: []string{"internal_host_for_ingress_1"}}},
+	}
+
+	err := caches.ValidateIngress(&translatedIngress)
+	if err == nil {
```

Comment:
I think we should check that this returns `ErrDomainConflict` otherwise we could return any other error and the test would pass anyway.

Reply:
There's no other error that could happen inside ValidateIngress /now/. But I will add that check in case we extend that validation function.